x-pack/solutions/observability/plugins/inventory/server/routes/entity_definition/get_entity_definitions.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { createInventoryServerRoute } from '../create_inventory_server_route';
+
+export const getEntityDefinitionSourceIndexPatternsByType = createInventoryServerRoute({
+  endpoint: 'GET /internal/inventory/entity/definitions/sources',
+  options: {
+    tags: ['access:inventory'],
+  },
+  async handler({ context, request, plugins }) {
+    const [_coreContext, entityManagerStart] = await Promise.all([
+      context.core,
+      plugins.entityManager.start(),
+    ]);
+    const entityManagerClient = await entityManagerStart.getScopedClient({ request });
+
+    const entityDefinitionsSource = await entityManagerClient.v2.readSourceDefinitions({});
+
+    const allEntityDefinitionIndexPatterns = entityDefinitionsSource.map((source) => ({
```

Comment:
this would do the same computation for every types with more than one source. would a reduce building a `{ type: index_patterns, .. }` object lower complexity ?

Reply:
Thanks for the suggestion! I thought that it wasn't readable before that's why I changed the reduce here. I tried to improve the `reduce` logic and added a test for it so it's easier to read and also covers some multi-source scenarios [here](https://github.com/elastic/kibana/pull/204026/commits/68d076c53ec315800c3f23c15f2b1bbbca33904e#diff-d724923f542eb5b4a1cd2a4866191d9c6552c42a4b9f2103d6388911ecc60b32). Is it better now, wdyt?

Reply:
lgtm, left a suggestion